proposals/kroxylicious-operator-api-alternative.md

```diff
@@ -0,0 +1,654 @@
+# Kroxylicious Operator API - Alternative Proposal.
+
+Keith isn't sure that 'Gateway API flavoured' approach for modelling ingress is the right one for Kroxylicious Operator.
+He thinks that our ingress API should work with Gateway APIs, rather than be one itself.  So, this 
+has prompted him to think about an alternative API design.
+
+## CRDs
+
+* Proxy CR - an instance of the Kroxylicious
+* ProxyIngress CR - Defines a way to access a Proxy
+* VirtualCluster CR - a virtual cluster
+* Filter CR - a filter definition
+
+![image](https://github.com/user-attachments/assets/8767da6b-3fc4-4fbc-99bc-64e93c06bc70)
+
+https://excalidraw.com/#json=CVAZMl-MuN1QP2cIW3hcA,cEWi5VjUg1mK2N1lvU-_xQ
+
+
+## Personas / Responsibilies
+
+* Infrastructure admin - responsible for the proxy tier and how proxies are exposed to the network.
+* Developer - responsible for the virtualcluster and filter
+
+## API
+
+### Proxy
+
+The Proxy CR is the responsibilty of the the Infrastructure admin.
+
+Declares an instance of the proxy and defines the ingress mechanisms available to it.
+
+```yaml
+apiVersion: proxy.kroxylicious.io/v1
```

Comment:
I appreciate  this is just a sketch but we should be starting with an `v1alpha1` version designator